scripts/img2img.py

```diff
@@ -241,8 +249,10 @@ def main():
     print(f"target t_enc is {t_enc} steps")
 
     precision_scope = autocast if opt.precision == "autocast" else nullcontext
+    if device.type == 'mps':
+        precision_scope = nullcontext # have to use f32 on mps
     with torch.no_grad():
-        with precision_scope("cuda"):
+        with precision_scope(device.type):
```

Comment:
This is 90% of the way to being able to inference on CPU, however you will get a type mismatch error, expecting a BFloat16 but getting a Float.

To fix this, it should be more like `if device.type in ['mps', 'cpu']:`
because CPUs don't have BFloat16 operations either.
I am not sure if this is a bug in PyTorch or not, because my impression is that `autocast()` should not be offering BFloat16 optimizations to anything that is not a recent Nvidia GPU.

Reply:
Thank you for sharing the solution 🙇

Reply:
> Thank you for sharing the solution 🙇 I was running `txt2img.py` and it kept failing with:
> 
> ```
> `RuntimeError: expected scalar type BFloat16 but found Float`
> ```
> 
>  
> But adding `--precision full` when running the script worked 🎉

Wow this perfectly solves my same problem.